### ISSUE TITLE
[BUG] TypedDict defined failed on module initialization (Ubuntu Py3.9)

### ISSUE THREAD

**Issue by KmolYuan** [cython/cython#4003]
**Describe the bug**
`TypeError` raised when defining a `TypedDict` inheritance in Ubuntu Python 3.9.

This error is displayed at runtime import instead of compile time.

**To Reproduce**
Code to reproduce the behaviour:
```cython
from typing import TypedDict

class A(TypedDict):
    pass
# TypeError: function() argument 'code' must be code, not str
class A(TypedDict, total=False):
    pass
# TypeError: 'total' is an invalid keyword argument for function()
```

**Expected behavior**
Import successfully.

**Environment (please complete the following information):**
 - OS: Ubuntu 16.04, 18.04, 20.04
 - Python version 3.9.1
 - Cython version 0.29.21, 3a6

**Comment by da-woods** [comment]
Duplicate of https://github.com/cython/cython/issues/3537

It isn't completely obvious that this is a duplicate, but `TypedDict` looks to use `__mro_entries__`. Once that's implemented I'd expect your code to work too. See https://github.com/cython/cython/issues/2753#issuecomment-777822900 for a workaround that you can probably adapt in the interim.

**Comment by da-woods** [comment]
It's fixed in the Cython 3 alpha. I don't think we're planning to back port the fix to Cython 0.29.x. The linked issue has some suggestions for possible workarounds
